### PR TITLE
Fix package aliases to follow golang convention

### LINKED ISSUE #35070: CRI: Package aliases are not following golang convention
As is mentioned in https://github.com/kubernetes/kubernetes/pull/33988#discussion_r83764898.

In CRI and dockershim, we are using package name containing capitalized characters e.g. `internalApi` and `runtimeApi`.

This is not align with golang convention https://blog.golang.org/package-names, we should change this to `internalapi` and `runtimeapi` in one PR.

@timstclair @yujuhong @feiskyer 
/cc @kubernetes/sig-node

### PR DESCRIPTION
**Pull request by feiskyer**
Some package aliases are not not align with golang convention https://blog.golang.org/package-names. This PR fixes them. Also adds a verify script and presubmit checks.

Fixes #35070.

cc/ @timstclair @Random-Liu

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35258)

### CODE REVIEW

**Comment by timstclair** [comment]
Thanks! LGTM.

/cc @kubernetes/sig-cluster-federation 

> By the way, it's better to add a presubmit check for this, but I didn't find a easy tool. Any ideas?

I think a simple grep check should be sufficient:
```
git grep -E $'^(import |\t)[a-z]+[A-Z_][a-zA-Z]* "[^"]+"$' -- '**/*.go' ':(exclude)vendor/*' ':(exclude)staging/*'
```

**Comment by feiskyer** [comment]
@timstclair Added presubmit hooks and rebased. PTAL.

**Comment by yujuhong** [comment]
This is going to cause a lot of churn for other PRs. I am not sure now is the best time.

**Comment by Random-Liu** [comment]
> This is going to cause a lot of churn for other PRs. I am not sure now is the best time.

Agree.

**Comment by timstclair** [comment]
When would be a better time? After code freeze, but before the branch is cut? Or after master is re opened? Neither seem like great times...

We could change the precommit check to only look at additions, rather than all files, which would prevent new uses of bad aliases. Either way, the check should probably be a submit requirement (e.g. `hack/verify-pkg-names.sh`). I don't know how many developers use the pre-commit hook.

**Comment by yujuhong** [comment]
> When would be a better time? After code freeze, but before the branch is cut? Or after master is re opened? Neither seem like great times...

I like cleanup in general, but I don't think this is urgent. There may not be a perfect time, but IMO, the beginning of a release cycle is better than two weeks before code freeze. That means after cutting the release branch, or even after the the release is out the door. This is just my two cents though.

**Comment by feiskyer** [comment]
@timstclair @yujuhong Agree it's not a good time now, I think 'after code freeze' is a good time for this.

**Comment by feiskyer** [comment]
Rebased. @timstclair do you think it's time for this now?

File: hack/verify-pkg-names.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+! files=$(git grep -E $'^(import |\t)[a-z]+[A-Z_][a-zA-Z]* "[^"]+"$' -- '**/*.go' ':(exclude)vendor/*' ':(exclude)staging/*')
```
Comment (timstclair) [inline hack/verify-pkg-names.sh]: Won't errexit terminate the script at this line if packages break the convention?

An alternative is to just inline it:
```
if git grep -E $'^(import |\t)[a-z]+[A-Z_][a-zA-Z]* "[^"]+"$' -- '**/*.go' ':(exclude)vendor/*' ':(exclude)staging/*'; then
...
```

File: hooks/pre-commit [hunk 2]
```diff
@@ -62,6 +62,20 @@ else
 fi
 echo "${reset}"
 
+echo -ne "Checking for package aliases... "
+files_break_alias=($(git grep -E $'^(import |\t)[a-z]+[A-Z_][a-zA-Z]* "[^"]+"$' -- '**/*.go' ':(exclude)vendor/*' ':(exclude)staging/*'
```
Comment (timstclair) [inline hooks/pre-commit]: Just call `hack/verify-pkg-names.sh` from here instead

**Comment by timstclair** [comment]
I think now is as good a time as any, as long as there's general approval for this convention. This will cause some rebase churn, but hopefully not too much.

/cc @lavalamp (buildcop)

File: hack/verify-pkg-names.sh (hunk 1 above)
Comment (feiskyer) [inline hack/verify-pkg-names.sh]: Yes, it exits in the following if-block. Will change to the alternative way.

File: hooks/pre-commit (hunk 2 above)
Comment (feiskyer) [inline hooks/pre-commit]: ack

**Comment by feiskyer** [comment]
@timstclair @lavalamp Addressed comments. PTAL.

**Comment by timstclair** [comment]
LGTM. Good luck with the rebases!

**Comment by feiskyer** [comment]
> Good luck with the rebases!

Yeah, too much rebases on this. @timstclair could you set a higher priority for this.

**Comment by timstclair** [comment]
Bumping to p2 to avoid rebase-hell

**Pull request merged**